proof/refine/ARM/IpcCancel_R.thy

```diff
@@ -2336,74 +2336,12 @@ lemma in_send_ep_queue_TCBBlockedSend:
   apply (fastforce simp: state_refs_of_def)
   done
 
-lemma in_recv_ep_queue_TCBBlockedRecv:
-  "\<lbrakk>kheap s epptr = Some (Endpoint (Structures_A.RecvEP queue)); t \<in> set queue; invs s\<rbrakk>
-   \<Longrightarrow> (epptr, TCBBlockedRecv) \<in> state_refs_of s t"
-  apply (prop_tac "valid_ep (Structures_A.RecvEP queue) s")
-   apply (fastforce simp: valid_objs_def valid_obj_def dest!: invs_valid_objs)
-  apply (clarsimp simp: state_refs_of_def valid_ep_def split: option.splits)
-  apply (intro conjI impI allI; (fastforce simp: obj_at_def)?)
-  apply (prop_tac "(t, EPRecv) \<in> state_refs_of s epptr", clarsimp simp: state_refs_of_def)
-  apply (clarsimp simp: sym_refs_def dest!: invs_sym_refs)
-  apply (fastforce simp: state_refs_of_def)
-  done
+(* FIXME : Move *)
+lemma pred_tcb_at_disj:
+  "(pred_tcb_at proj (\<lambda>x. P x \<or> P' x) p s) = (pred_tcb_at proj (\<lambda>x. P x) p s \<or> pred_tcb_at proj (\<lambda>x. P' x) p s)"
```

Comment:
Could we quickly do this move now? Or at least to Move_R?

Reply:
Surely... and what I discovered is that we actually have this in Invariants_AI:
```
lemma pred_tcb_at_disj:
  "(pred_tcb_at proj P t s ∨ pred_tcb_at proj Q t s) = pred_tcb_at proj (λa. P a ∨ Q a) t s"
  by (fastforce simp add: pred_tcb_at_def obj_at_def)
```
My natural thinking would go in the way of splitting `pred_tcb_at` with the disjunction, but people have different ideas apparently...

Reply:
And in its only use in AInvs is with `symmetric`... so I might as well flip it :D